Tests/CommandsTests/PackageCommandTests.swift

```diff
@@ -875,109 +880,178 @@ class PackageCommandTestCase: CommandsBuildProviderTestCase {
+    func executeAddURLDependencyAndAssert(
+        packagePath: AbsolutePath,
+        initialManifest: String? = nil,
+        url: String,
+        requirementArgs: [String],
+        expectedManifestString: String,
+    ) async throws {
+        _ = try await execute(
+            ["add-dependency", url] + requirementArgs,
+            packagePath: packagePath,
+            manifest: initialManifest
+        )
+        try assertManifestContains(packagePath, expectedManifestString)
+    }
+
+    func testPackageAddDifferentDependencyWithSameURLTwiceFails() async throws {
+        try await testWithTemporaryDirectory { tmpPath in
+            let fs = localFileSystem
+            let path = tmpPath.appending("PackageB")
+            try fs.createDirectory(path)
+
+            let url = "https://github.com/swiftlang/swift-syntax.git"
+            let manifest = """
+                // swift-tools-version: 5.9
+                import PackageDescription
+                let package = Package(
+                    name: "client",
+                    dependencies: [
+                        .package(url: "\(url)", exact: "601.0.1")
+                    ],
+                    targets: [ .target(name: "client", dependencies: [ "library" ]) ]
+                )
+            """
+
+            try localFileSystem.writeFileContents(path.appending("Package.swift"), string: manifest)
+
+            await XCTAssertThrowsCommandExecutionError(
+                try await execute(["add-dependency", url, "--revision", "58e9de4e7b79e67c72a46e164158e3542e570ab6"], packagePath: path)
+            ) { error in
+                XCTAssertMatch(error.stderr, .contains("error: unable to add dependency 'https://github.com/swiftlang/swift-syntax.git' because it already exists in the list of dependencies"))
+            }
+        }
+    }
+
+    func testPackageAddSameDependencyTwiceHasNoEffect() async throws {
+        try await testWithTemporaryDirectory { tmpPath in
+            let fs = localFileSystem
+            let path = tmpPath.appending("PackageB")
+            try fs.createDirectory(path)
+
+            let url = "https://github.com/swiftlang/swift-syntax.git"
+            let manifest = """
+                // swift-tools-version: 5.9
+                import PackageDescription
+                let package = Package(
+                    name: "client",
+                    dependencies: [
+                        .package(url: "\(url)", exact: "601.0.1"),
+                    ],
+                    targets: [ .target(name: "client", dependencies: [ "library" ]) ]
+                )
+            """
+            let expected = #".package(url: "https://github.com/swiftlang/swift-syntax.git", exact: "601.0.1"),"#
+
+            try await executeAddURLDependencyAndAssert(
+                packagePath: path,
+                initialManifest: manifest,
+                url: "https://github.com/swiftlang/swift-syntax.git",
```

Comment:
question: Does SwiftPM treat a package dependency of `"https://github.com/swiftlang/swift-syntax.git"` the same as `"https://github.com/swiftlang/swift-syntax"`?  Is there value if documenting this specific case?

what about repositories that has the same name, but belong to different organizations/users?